cost/google/unattached_volumes/google_delete_unattached_volumes.pt

```diff
@@ -0,0 +1,388 @@
+  description "The number of days a volume has been unattached"
+  default 30
+end
+
+parameter "param_exclusion_tags" do
+  label "Exclusion Label List"
+  description "Cloud native label to ignore volumes. Format: Key:Value"
+  type "list"
+  allowed_pattern /(^$)|([\w]?)+\:([\w]?)+/
+end
+
+parameter "param_create_final_snapshot" do
+  label "Create Final Snapshot"
+  description "Boolean for whether or not to take a final snapshot before deleting"
+  type "string"
+  allowed_values "true", "false"
+end
+
+
+###############################################################################
+# Authentication
+###############################################################################
+
+auth "auth_google", type: "oauth2" do
+  token_url "https://www.googleapis.com/oauth2/v4/token"
+  grant type: "jwt_bearer" do
+    iss cred("GCE_PLUGIN_ACCOUNT")
+    aud "https://www.googleapis.com/oauth2/v4/token"
+    additional_claims do {
+      "scope" => "https://www.googleapis.com/auth/compute",
+      "scope" => "https://www.googleapis.com/auth/cloud-platform"
+    } end
+    signing_key cred("GCE_PLUGIN_PRIVATE_KEY")
+  end
+end
+
+###############################################################################
+# Pagination
+###############################################################################
+
+pagination "google_pagination" do
+  get_page_marker do
+    body_path "nextPageToken"
+  end
+  set_page_marker do
+    query "pageToken"
+  end
+end
+
+###############################################################################
+# Datasources
+###############################################################################
+
+datasource "ds_projects" do
+  request do
+    auth $auth_google
+    host "cloudresourcemanager.googleapis.com"
+    verb "GET"
+    path "/v1/projects"
+    pagination $google_pagination
+  end
+  result do
+    encoding "json"
+    collect jmes_path(response,"projects[*]") do
+      field "project_name", jmes_path(col_item, "projectId")
+      end
+  end
+end
+
+datasource "ds_zones" do
+  iterate $ds_projects
+  request do
+    auth $auth_google
+    host "compute.googleapis.com"
+    verb "GET"
+    path join(["/compute/v1/projects/", val(iter_item, "project_name"), "/zones"])
+    query "project", val(iter_item, "project_name")
+    pagination $google_pagination
+  end
+  result do
+    encoding "json"
+    collect jmes_path(response,"items[*]") do
+      field "region", last(split(jmes_path(col_item, "region"),'/'))
+      field "zone", jmes_path(col_item, "name") 
+    end
+  end
+end
+
+datasource "ds_merged_project_zone" do
+  run_script $js_merge, $ds_zones, $ds_projects
+end
+
+script "js_merge", type: "javascript" do
+  parameters "ds_zones", "ds_projects"
+  result "res"
+  code <<-EOS
+  res = []
+  _.each(ds_projects, function(project){
+    _.each(ds_zones, function(zone){
+      var result_json = {"project": project.project_name,"zone": zone.zone}
+      res.push(result_json)
+    })
+  })
+  EOS
+end
```

Comment:
I'm not sure this is really needed.  please evaluate and remove if it's not necessary.